sources/Application/Views/FieldView.cpp

```diff
@@ -127,8 +127,19 @@ void FieldView::ProcessButtonMask(unsigned short mask, bool pressed) {
                 if (next) {
                   if ((*it)->GetPosition()._y < next->GetPosition()._y) {
                     next = *it;
-                  } else {
-                    // if both target at same height
+                  } else if ((*it)->GetPosition()._y ==
+                             next->GetPosition()._y) {
+                    // if both targets at same height, prefer the target with an
+                    // X value closest to the current focus
+
+                    // cast to signed ints
+                    int32_t itX = (*it)->GetPosition()._x;
```

Comment:
one small question: do these need to be signed as the screen "character cell" coordinates that we use should always be positive and even should always be <256.
I guess the GuiPoint's use of `long`'s we inherited from LGPT kind of obscures this and these are only local variables so not really an issue, just thought I'd ask.

Reply:
yeah so the reason i'm making them signed values here is that I want a signed result from the subtraction right below. Like, `abs(1 - 2)` should be `1`, not `255`.

C++'s automatic number conversion rules are not something I really have internalized, so I'm not sure whether subtraction automatically makes the result be a signed value even if the inputs are unsigned values. So I figured I'd do it this way to be safe.